include/simdutf/internal/isadetection.h

```diff
@@ -84,6 +88,41 @@ static inline uint32_t detect_supported_architectures() {
   return instruction_set::ALTIVEC;
 }
 
+#elif SIMDUTF_IS_RISCV64
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <asm/hwprobe.h>
+#include <sys/syscall.h>
+#endif
+
+static inline uint32_t detect_supported_architectures() {
+  uint32_t host_isa = instruction_set::DEFAULT;
+#if SIMDUTF_IS_RVV
+  host_isa |= instruction_set::RVV;
+#endif
+#if SIMDUTF_IS_ZVBB
+  host_isa |= instruction_set::ZVBB;
+#endif
+#if defined(__linux__) && defined(RISCV_HWPROBE_IMA_V)
+  riscv_hwprobe probes[] = {
+    { RISCV_HWPROBE_IMA_V, 0 },
+#ifdef RISCV_HWPROBE_EXT_ZVBB
+    { RISCV_HWPROBE_EXT_ZVBB, 0 },
+#endif
+  };
+  syscall(__NR_riscv_hwprobe,
```

Comment:
Shouldn't we test result of syscall is zero before testing `probes` array?

Reply:
Yeah, that's true. I think I also misunderstood how the hwprobe api works, I'll change it to the way the [RISC-V optimization guide](https://riscv-optimization-guide.riseproject.dev/) recommends.